lib/blocs/home_page_cubit.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:convert';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:trufi_core/models/map_route_state.dart';
+import 'package:trufi_core/repository/local_repository.dart';
+import 'package:trufi_core/repository/request_manager.dart';
+import 'package:trufi_core/trufi_models.dart';
+
+class HomePageBloc extends Cubit<MapRouteState> {
+  LocalRepository localRepository;
+  RequestManager requestManager;
+
+  HomePageBloc(this.localRepository)
+      : super(MapRouteState(
+          isFetching: false,
+          showSuccessAnimation: false,
+        ));
+
+  Future<void> loadFromSharedPreferences() async {
```

Comment:
We need to call this method for load this Last-MapRouteState, (when we write the MapRouteState with localRepository this MapRouteState saved)) 

```
  Map<String, dynamic> toJson() {
    return {
      _fromPlace: fromPlace?.toJson(),
      _toPlace: toPlace?.toJson(),
      _plan: plan?.toJson(),
      _ad: ad?.toJson(),
    };
  }
```

 isFetching:null and showAnimation: null, this value do error in PlanPageState

Solution:
1)call method in constructor.

```
  HomePageBloc(this.localRepository)
      : super(MapRouteState(
          isFetching: false,
          showSuccessAnimation: false,
        )) {
    loadFromSharedPreferences();
  }
```

2) We don't called _fetch(method in HomePage) then We define defectValue in MapRouteState.

```
 MapRouteState({
    this.fromPlace,
    this.toPlace,
    this.plan,
    this.ad,
    this.isFetching = false,
    this.showSuccessAnimation = false,
    this.currentFetchAdOperation,
    this.currentFetchPlanOperation,
  }); 
```

or  call loadFromSharedPrefecrences with (CopyWith)
```
  Future<void> loadFromSharedPreferences() async {
    final jsonString = await localRepository.getStateHomePage();

    if (jsonString != null && jsonString.isNotEmpty) {
      emit(
        MapRouteState.fromJson(jsonDecode(jsonString) as Map<String, dynamic>)
        .copyWith(
          isFetching: false,
          showSuccessAnimation: false,
        ),
      );
    }
  }
```

Reply:
Fantastic point, thank you for the catch. I changed the toJson method slightly so we have now a possibility to save the null values to false. This way we can assure that there should never be a null state in the DB. I guess that would be the cleanest way.

I also tried to use https://pub.dev/packages/json_serializable the only problem for this class are `currentFetchAdOperation` and `currentFetchPlanOperation`. If we can remove them later we could remove the whole toJson / fromJson and let it handle by the build runner.

Reply:
it's a great idea to use  https://pub.dev/packages/json_serializable, What do you think of https://pub.dev/packages/hydrated_bloc?

Reply:
That would be even better I guess. 👍 But I would do it in the next PR.

Reply:
Good :), jejeje I could add code(feats/issues) after this PR ends 👍